### PR TITLE
Vmray new funcs

### PR DESCRIPTION
**Pull request by aburt-demisto**
## Status
Ready/In Progress/In Hold(Reason for hold)

## Related Issues
fixes: link to the issue

## Description
A few sentences describing the overall goals of the pull request's commits.

## Screenshots
Paste here any images that will help the reviewer

## Related PRs
List related PRs against other branches:

branch | PR
------ | ------

## Required version of Demisto
x.x.x

## Does it break backward compatibility?
   - Yes
       - Further details:
   - No

## Must have
- [ ] Tests
- [ ] Documentation (with link to it)
- [ ] Code Review

## Dependencies
Mention the dependencies of the entity you changed as given from the precommit hooks in checkboxes, and tick after tested them.
- [ ] Dependency 1
- [ ] Dependency 2
- [ ] Dependency 3

## Additional changes
Describe additional changes done, for example adding a function to common server.

## Technical writer review
Mention and link to the files that require a technical writer review.
- [ ] [YAML file](link)
- [ ] [CHANGELOG](link)

### CODE REVIEW

**Comment by aburt-demisto** [comment]
@Itay4 I have merged this pull request with the previous for the 2 new functions. I have renamed the 'file' function to be 'vmray_check_file'. There is also 'check_file_command' that supports the 'vmray-check-file' command.

File: Integrations/VMRay/VMRay.yml [hunk 1]
```diff
@@ -1,6 +1,6 @@
 category: Forensics & Malware Analysis
 commonfields:
-  id: vmray
+  id: vmray v2
```
Comment (Itay4) [inline Integrations/VMRay/VMRay.yml]: please revert this

File: Integrations/VMRay/VMRay.yml [hunk 2]
```diff
@@ -22,7 +22,7 @@ configuration:
   type: 8
 description: Ransomware analysis sandboxing.
 display: VMRay
-name: vmray
+name: vmray v2
```
Comment (Itay4) [inline Integrations/VMRay/VMRay.yml]: please revert this

File: Integrations/VMRay/VMRay.yml [hunk 3]
```diff
@@ -519,9 +519,23 @@ script:
...
+    name: vmray-check-file
   dockerimage: demisto/python
   isfetch: false
+  longRunning: false
+  longRunningPort: false
   runonce: false
-  script: ''
+  script: '-'
   type: python
-fromversion: 4.0.0
```
Comment (Itay4) [inline Integrations/VMRay/VMRay.yml]: please revert this

File: Integrations/VMRay/VMRay.yml [hunk 4]
```diff
@@ -519,9 +519,23 @@ script:
...
+sourcemoduleid: vmray
```
Comment (Itay4) [inline Integrations/VMRay/VMRay.yml]: why did you add this?

File: Integrations/VMRay/VMRay.py [hunk 5]
```diff
@@ -490,6 +490,56 @@ def get_submission(submission_id):
     response = http_request('GET', url_suffix=suffix)
     return response
 
+def check_file_command():
+    file_hash_input = demisto.args().get('file')
+    file_hashes = file_hash_input.split(',')
```
Comment (Itay4) [inline Integrations/VMRay/VMRay.py]: please use `argToList`

File: Integrations/VMRay/VMRay.py [hunk 6]
```diff
@@ -490,6 +490,56 @@ def get_submission(submission_id):
...
+    for file_hash in file_hashes:
+        if len(file_hash) != 32 and len(file_hash) != 40 and len(file_hash) != 64:
```
Comment (Itay4) [inline Integrations/VMRay/VMRay.py]: please use `get_hash_type`

File: Integrations/VMRay/VMRay.py [hunk 7]
```diff
@@ -490,6 +490,56 @@ def get_submission(submission_id):
...
+        raw_response = check_file(file_hash, hash_type)
+        data = raw_response.get('data')
+
+        if len(data) == 0:
+            return_outputs(
+                'No submission found in VMRay for hash : {}'.format(file_hash),
+                {},
+            )
+
+        for item in data:
```
Comment (Itay4) [inline Integrations/VMRay/VMRay.py]: this should be in `else` clause of the condition above

File: Integrations/VMRay/VMRay.py [hunk 8]
```diff
@@ -490,6 +490,56 @@ def get_submission(submission_id):
...
+        hash_type = 'md5'
+        if len(file_hash) == 40:
+            hash_type = 'sha1'
+        if len(file_hash) == 64:
+            hash_type = 'sha256'
+
+        raw_response = check_file(file_hash, hash_type)
+        data = raw_response.get('data')
+
+        if len(data) == 0:
```
Comment (Itay4) [inline Integrations/VMRay/VMRay.py]:
```suggestion
        if not data:
```

File: Integrations/VMRay/VMRay.py [hunk 9]
```diff
@@ -490,6 +490,56 @@ def get_submission(submission_id):
...
+            return_outputs(
+                'No submission found in VMRay for hash : {}'.format(file_hash),
```
Comment (Itay4) [inline Integrations/VMRay/VMRay.py]: we should return dbot score entry here as well, with `Unknown` verdict

File: Integrations/VMRay/VMRay.py [hunk 10]
```diff
@@ -490,6 +490,56 @@ def get_submission(submission_id):
...
+                'VMRay.Sample(var.SampleID === obj.SampleID)': entry,
+                outputPaths.get('dbotscore'): scores,
+            }
+
+            human_readable = tableToMarkdown(
+                'Results for sample id: {} with severity {}'.format(
+                    entry.get('SampleID'), entry.get('Severity')
+                ),
+                entry,
+                headers=['Type', 'MD5', 'SHA1', 'SHA256', 'SSDeep'],
```
Comment (Itay4) [inline Integrations/VMRay/VMRay.py]: please remove trailing slash

File: Integrations/VMRay/VMRay.py [hunk 11]
```diff
@@ -490,6 +490,56 @@ def get_submission(submission_id):
...
+            entry['SSDeep'] = item.get('sample_ssdeephash')
+            entry['Severity'] = SEVERITY_DICT.get(item.get('sample_severity'))
+            entry['Type'] = item.get('sample_type')
+            entry['Created'] = item.get('sample_created')
+            entry['Classification'] = item.get('sample_classifications')
+            scores = dbot_score_by_hash(entry)
+
+            entry_context = {
+                'VMRay.Sample(var.SampleID === obj.SampleID)': entry,
+                outputPaths.get('dbotscore'): scores,
```
Comment (Itay4) [inline Integrations/VMRay/VMRay.py]: please remove trailing slash

File: Integrations/VMRay/VMRay.py [hunk 12]
```diff
@@ -490,6 +490,56 @@ def get_submission(submission_id):
...
+            entry['SHA1'] = item.get('sample_sha1hash')
+            entry['SHA256'] = item.get('sample_sha256hash')
+            entry['SSDeep'] = item.get('sample_ssdeephash')
+            entry['Severity'] = SEVERITY_DICT.get(item.get('sample_severity'))
+            entry['Type'] = item.get('sample_type')
+            entry['Created'] = item.get('sample_created')
+            entry['Classification'] = item.get('sample_classifications')
+            scores = dbot_score_by_hash(entry)
+
+            entry_context = {
```
Comment (Itay4) [inline Integrations/VMRay/VMRay.py]: need also to return to `File` path according to [context standard](https://github.com/demisto/content/tree/master/docs/context_standards#file)

File: Integrations/VMRay/VMRay.py (hunk 5 above)
Comment (aburt-demisto) [reply Integrations/VMRay/VMRay.py]: I haven't modified this from the original code. I will leave this for someone else to look at.

File: Integrations/VMRay/VMRay.py (hunk 6 above)
Comment (aburt-demisto) [reply Integrations/VMRay/VMRay.py]: Done

File: Integrations/VMRay/VMRay.py (hunk 9 above)
Comment (aburt-demisto) [reply Integrations/VMRay/VMRay.py]: As this is no longer providing the 'file' functionality, I don't think it should return any DBot Score. The lookup function is to find out whether a submission has already occurred for the sample, not to look up it's value. The lookup for the sample can then be acheived using the returned samplID from the lookup function.

File: Integrations/VMRay/VMRay.py (hunk 7 above)
Comment (aburt-demisto) [reply Integrations/VMRay/VMRay.py]: Done

File: Integrations/VMRay/VMRay.py (hunk 12 above)
Comment (aburt-demisto) [reply Integrations/VMRay/VMRay.py]: This part of the function was taken straight from the get_sample_command() function.

File: Integrations/VMRay/VMRay.py (hunk 10 above)
Comment (aburt-demisto) [reply Integrations/VMRay/VMRay.py]: Which trailing slash?

File: Integrations/VMRay/VMRay.yml (hunk 1 above)
Comment (aburt-demisto) [reply Integrations/VMRay/VMRay.yml]: Done

File: Integrations/VMRay/VMRay.yml (hunk 2 above)
Comment (aburt-demisto) [reply Integrations/VMRay/VMRay.yml]: Done

File: Integrations/VMRay/VMRay.yml (hunk 3 above)
Comment (aburt-demisto) [reply Integrations/VMRay/VMRay.yml]: Done

File: Integrations/VMRay/VMRay.yml (hunk 4 above)
Comment (aburt-demisto) [reply Integrations/VMRay/VMRay.yml]: I didn't edit the original code as I couldn't import it to test, so had to copy the original code into a "v2" integration.

**Comment by aburt-demisto** [comment]
@Itay4 I've completed most of the changes (I believe)

**Comment by Itay4** [comment]
@aburt-demisto please note [the build](https://circleci.com/gh/demisto/content/31400) fails

**Comment by aburt-demisto** [comment]
I have not made any other changes. I do not have a "build" environment to test the code using PyCharm scripts as the environment does not work correctly in Windows. I am building an Ubuntu environment to support this in the future but do not have the time to test the code. There is only 1 function being added though. I can send that code spearately if needed, for the inclusion.

**Comment by anara123** [comment]
@aburt-demisto you can use our circle ci build in order to make sure your code pass all the validations.
Currently it is failing on
`Key 'sourcemoduleid' was not defined. Path: ''.: Path: '/'>` it means you need to delete this key 
The display name of the insecure parameter should be 'Trust any certificate (not secure)'` , etc.

**Comment by aburt-demisto** [comment]
No longer required
